File: pysrc/fractal/evolve/_evolving/_evolving.py
"""

    *Evolving*

"""

from dataclasses import dataclass

from fractal.brain import Brain

@dataclass
class Evolving:
    brain: Brain
    placements: Vector[TrainingSample]
    data: NamedTuple # (W, X, A, P, V) tuple obtained after converting `samples`
    Wmean: float
    Hp: float
    batches_stream: Iterator[Batch]

    end |> Util.cycle_iterator |> Iterators.Stateful
    return new(network, samples, params, data, Wmean, Hp, batches_stream,)

File: src/fractal/mind/evolve/_evolving/_evolving.rs

pub struct Evolving {
    brain: Brain,
    placements: Vec<TrainingSample>,
    # data: NamedTuple # (W, X, A, P, V) tuple obtained after converting `samples`,
    Wmean: f32,
    Hp: f32,
    batches_stream: Iterator<Batch>,
}

impl Evolving {
    # end |> Util.cycle_iterator |> Iterators.Stateful
    return new(network, samples, params, data, Wmean, Hp, batches_stream,)
    }